### PR TITLE
Implement multi launch angle simulations

### PR DESCRIPTION
**Pull request by marcelvanworkum**
Refactored directory structure,
Forced Loader to be synchronous
Perform the simulations and output
the result to a json hash string

### CODE REVIEW

File: core/src/net/sf/openrocket/database/AsynchronousDatabaseLoader.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
 	private volatile boolean startedLoading = false;
 	private volatile boolean endedLoading = false;
 	private volatile boolean inUse = false;
+
+	private boolean forceSynchronousLoading = true;
```
Comment (jamiesanson) [inline core/src/net/sf/openrocket/database/AsynchronousDatabaseLoader.java:16]: Should we bother with fixing/updating the Java docs around changes made in this file?

File: core/src/net/sf/openrocket/simulation/FlightData.java [hunk 2]
```diff
@@ -310,4 +308,42 @@ private double calculateMaxAcceleration() {
...
 	}
+
+	/////////////////////
+	///  TEAM ROCKET  ///
+	/// Modifications ///
+	/////////////////////
+
+	private double launchAngle;
+
+    /**
```
Comment (jamiesanson) [inline core/src/net/sf/openrocket/simulation/FlightData.java]: Indentation should match rest of file

File: core/src/net/sf/openrocket/simulation/SimulationOptions.java [hunk 3]
```diff
@@ -403,6 +403,10 @@ public void setRandomSeed(int randomSeed) {
 		 */
 		//		fireChangeEvent();
 	}
+
+	public double getMaxLaunchRodAngle(){
+		return this.MAX_LAUNCH_ROD_ANGLE;
```
Comment (jamiesanson) [inline core/src/net/sf/openrocket/simulation/SimulationOptions.java]: Might be better to have this as a `public static final` field, so when using it we can just use `SimulationOptions.MAX_LAUNCH_ROD_ANGLE`

File: teamrocket/src/TeamRocket.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+    /**
+     * Executes the simulations
+     *
+     * TODO this should take in a variable of config params so that we can run different types of simulations/pass in weather/gps location
+     *
+     * @return JSON string containing simulation results
+     */
+	public String runSimulations() {
```
Comment (jamiesanson) [inline teamrocket/src/TeamRocket.java:49]: Inconsistent indentation in this and following method

File: teamrocket/src/TeamRocket.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+			System.err.println(e.getMessage());
+		}
+
+		return document;
+	}
+
+	/*
+	    This is very hacky, but essentially is instantiates all of the dependencies needed to make OpenRocket function
+	 */
+	private void dangerouslyInstantiateOpenRocketDependencies() {
```
Comment (jamiesanson) [inline teamrocket/src/TeamRocket.java:89]: Spooked

File: teamrocket/src/util/JSONFormatter.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                stringifiedSimulationData += "} ";
+            }
+        }
+
+        stringifiedSimulationData += "}";
+
+        return stringifiedSimulationData;
+    }
+
+    private static String ConvertFlightData(ArrayList<FlightData> flightDataList) {
```
Comment (jamiesanson) [inline teamrocket/src/util/JSONFormatter.java]: camelCase not PascalCase for method names

File: teamrocket/src/util/SimulationRunner.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Utility class to perform a variety of openrocket simulations
+ *
+ * Created 3rd October 2016
+ * By Marcel van Workum
+ */
+public class SimulationRunner {
+    private OpenRocketDocument openRocketDocument;
+    private SimulationListener simulationListener = new AbstractSimulationListener();
```
Comment (jamiesanson) [inline teamrocket/src/util/SimulationRunner.java]: `private final`

**Review by jamiesanson — CHANGES_REQUESTED** [review]
Overall looks good. Just minor things like indentation consistency and field declarations etc.

File: core/src/net/sf/openrocket/database/AsynchronousDatabaseLoader.java (hunk 1 above)
Comment (marcelvanworkum) [reply core/src/net/sf/openrocket/database/AsynchronousDatabaseLoader.java:16]: Yeah should probably keep things consistent. I'll do that.

File: teamrocket/src/TeamRocket.java (hunk 5 above)
Comment (marcelvanworkum) [reply teamrocket/src/TeamRocket.java:89]: Ha yeah, but I am somewhat following the convention of ReactJs's #dangerouslySetInnerHtml

https://facebook.github.io/react/tips/dangerously-set-inner-html.html

File: core/src/net/sf/openrocket/simulation/SimulationOptions.java (hunk 3 above)
Comment (marcelvanworkum) [reply core/src/net/sf/openrocket/simulation/SimulationOptions.java]: Agreed

File: teamrocket/src/TeamRocket.java (hunk 5 above)
Comment (jamiesanson) [reply teamrocket/src/TeamRocket.java:89]: "is intentionally chosen to be frightening". Fair enough

File: core/src/net/sf/openrocket/simulation/FlightData.java (hunk 2 above)
Comment (marcelvanworkum) [reply core/src/net/sf/openrocket/simulation/FlightData.java]: That's odd, in IntelliJ these are indented correctly... I will try auto-indent the whole file

File: core/src/net/sf/openrocket/simulation/SimulationOptions.java (hunk 3 above)
Comment (marcelvanworkum) [reply core/src/net/sf/openrocket/simulation/SimulationOptions.java]: Turns out it already was, and the getter wasn't even being used. Have removed.

File: teamrocket/src/TeamRocket.java (hunk 4 above)
Comment (marcelvanworkum) [reply teamrocket/src/TeamRocket.java:49]: Again, was showing as fine in IntelliJ. Will auto-indent.

File: core/src/net/sf/openrocket/simulation/SimulationOptions.java (hunk 3 above)
Comment (jamiesanson) [reply core/src/net/sf/openrocket/simulation/SimulationOptions.java]: Neato

File: teamrocket/src/TeamRocket.java (hunk 5 above)
Comment (marcelvanworkum) [reply teamrocket/src/TeamRocket.java:89]: :ghost:

**Comment by marcelvanworkum** [comment]
@jamiesanson back to you

**Comment by marcelvanworkum** [comment]
@jamiesanson unfortunately the auto-indent touches a bunch of lines and makes the commit look less clean. But I guess it makes sense, and it fixes the indentation issues.

**Review by jamiesanson — APPROVED** [review]

**Comment by jamiesanson** [comment]
Approved. Merging

**Pull request merged**